### PR TITLE
Add Github action workflow to update API docs

### PR DESCRIPTION
**Pull request by thardeck**
Refers to #https://github.com/rancher/fleet/issues/2673

The Github action workflow was run successfully against my fork: https://github.com/thardeck/fleet-docs/actions/workflows/update-api-docs.yml

The workflow is run every day and can be run manually.
It will never generate more than one pr at a time and does not create anything if nothing was changed.

### CODE REVIEW

File: .github/scripts/update-api.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    for file_path in "${RELEVANT_CRD_FILES[@]}"; do
+        [ -f "$file_path" ] && crd_files="$crd_files -f $file_path"
+    done
+
+    # shellcheck disable=SC2086
+    ../crd-to-markdown $crd_files \
+        -n Bundle -n BundleDeployment -n GitRepo -n GitRepoRestriction -n BundleNamespaceMapping -n Content \
+        -n Cluster -n ClusterRegistration -n ClusterRegistrationToken -n ClusterGroup -n ImageScan \
+        | sed -e 's/\[\]\[/\\[\\]\[/' \
```
Comment (0xavi0) [inline .github/scripts/update-api.sh:44]: Question: 
Shouldn't it be: 
```
s/\[\]\[/\\[\\]\\[/
```

For every `\[` it should add `\\[`... and the last one is just `\[`

Comment (0xavi0) [reply .github/scripts/update-api.sh:44]: Oh, replying to myself... 
This is to create the 
```
\[\][BundleResource]
``` 
output, for example. 

Looks good :)

**Review by 0xavi0 — COMMENTED** [review]
I think the `[Back to Custom Resources](#custom-resources)` link generated is broken.

As the script does:
`s/### Custom Resources/# Custom Resources Spec/ `

The correct output for that link should be:
`[Back to Custom Resources](#custom-resources-spec)`

At least that's what I got when running the script locally. 

I think we need an extra `sed` to change all occurrences of `#custom-resources` to `#custom-resources-spec`

File: .github/scripts/update-api.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        > "$ref_crd_path"
+}
+
+generate_markdown_files() {
+    pushd fleet || exit
+    git checkout main
+    create_markdown "../fleet-docs/docs/ref-crds.md"
+    popd || exit
+
+    for directory in ./fleet-docs/versioned_docs/*; do
```
Comment (weyfonk) [inline .github/scripts/update-api.sh:59]: Right, so this updates API docs for all versions. How likely are we to need this for versions which are no longer maintained though, eg. currently anything older than, say, 0.8?

File: .github/scripts/update-api.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    "pkg/apis/fleet.cattle.io/v1alpha1/bundledeployment_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/bundlenamespacemapping_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/cluster_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/clustergroup_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/clusterregistration_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/clusterregistrationtoken_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/content_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/gitrepo_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/gitreporestriction_types.go"
+    "pkg/apis/fleet.cattle.io/v1alpha1/imagescan_types.go"
```
[the remark this answers is not in the document]
Comment (weyfonk) [reply .github/scripts/update-api.sh:18]: I also think that an ignore list could be more future-proof; on the other hand, the command line on L42+ would need an update anyway if we were to add new CRDs... Although I'd be tempted to say that we don't add CRDs that often. Some upcoming features could change that, but this is already an improvement.

**Review by 0xavi0 — APPROVED** [review]
LGTM

**Pull request merged**